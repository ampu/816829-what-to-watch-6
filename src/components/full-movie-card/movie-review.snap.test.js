import React from 'react';

import {render} from '@testing-library/react';

import MovieReview from './movie-review';

describe(`MovieReview component`, () => {
  it(`Should render according to the snapshot with no review provided`, () => {
    const {container} = render(<MovieReview/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot with review provided`, () => {
    const review = {
      id: `fakeId`,
      user: {
        id: `fakeUserId`,
        name: `fakeUserName`,
      },
      rating: 9.9,
      text: `fakeText`,
      date: `2020-03-09`,
    };
    const {container} = render(<MovieReview review={review}/>);
    expect(container).toMatchSnapshot();
  });
});
