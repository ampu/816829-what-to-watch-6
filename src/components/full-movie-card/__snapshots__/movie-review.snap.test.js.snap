// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieReview component Should render according to the snapshot with no review provided 1`] = `
<div>
  <div
    class="review"
  >
    <blockquote
      class="review__quote"
    >
      
      <footer
        class="review__details"
      >
        
      </footer>
    </blockquote>
    0
  </div>
</div>
`;

exports[`MovieReview component Should render according to the snapshot with review provided 1`] = `
<div>
  <div
    class="review"
  >
    <blockquote
      class="review__quote"
    >
      <p
        class="review__text"
      >
        fakeText
      </p>
      <footer
        class="review__details"
      >
        <cite
          class="review__author"
        >
          fakeUserName
        </cite>
        <time
          class="review__date"
          datetime="2020-03-09"
        >
          March 09, 2020
        </time>
      </footer>
    </blockquote>
    <div
      class="review__rating"
    >
      9,9
    </div>
  </div>
</div>
`;
